src/smt/theory_str_noodler/formula.h

```diff
@@ -498,10 +506,80 @@ namespace smt::noodler {
             return ret;
         }
 
+        /**
+         * @brief Check whether a formula is quadratic.
+         * 
+         * @return true <-> quadratic
+         */
+        bool is_quadratic() const {
+            std::map<BasicTerm, unsigned> occur_map;
+
+            auto upd = [&occur_map] (const std::vector<BasicTerm>& vc) {
+                for(const BasicTerm& bt : vc) {
+                    if(!bt.is_variable()) continue;
+                    auto it = occur_map.find(bt);
+                    if(it != occur_map.end()) {
+                        it->second++;
+                    } else {
+                        occur_map.emplace(bt, 1);
+                    }
```

Comment:
You can just have `occur_map[bt]++` here, `unsigned` [should be value initialized, i.e. 0](https://stackoverflow.com/questions/5616421/increment-mapstring-int-using-operator)